src/main_context_channel.rs

```diff
@@ -224,96 +224,6 @@ struct ChannelSource<T, F: FnMut(T) -> Continue + 'static> {
     callback: Option<RefCell<F>>,
 }
 
-unsafe extern "C" fn prepare<T>(
```

Comment:
These should stay outside. They're not trampolines like the others ones, but the actual implementation of these functions. It's not just something translating between C and Rust.